talktorials/11_online_apis/11b_online_docking.ipynb

```diff
@@ -0,0 +1,823 @@
+    "    ligands.append(m)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Use SwissDock\n",
+    "\n",
+    "SwissDock uses a SOAP interface, so we will need to install `suds` for that.\n",
+    "\n",
+    "> Notice: SwissDock servers are not working lately. Go to the OPAL alternative below!"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {},
+   "source": [
+    "!pip install suds-community"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from suds.client import Client\n",
+    "import zlib\n",
+    "import string\n",
+    "import requests\n",
+    "\n",
+    "def swissdock_client():\n",
+    "    # Server seems to be down at the moment...\n",
+    "    # http://swissdock.vital-it.ch/soap/ replies with 503 Unavailable\n",
+    "    # because it points to wrong domain... patch it?\n",
+    "    SWISSDOCK_WSDL_URL = \"http://www.swissdock.ch/soap/wsdl\"\n",
+    "    r = requests.get(\"http://www.swissdock.ch/soap/wsdl\")\n",
+    "    r.raise_for_status()\n",
+    "    WSDL = r.text.replace(\"http://swissdock.vital-it.ch/soap/\", \"http://www.swissdock.ch/soap/\")\n",
+    "    with open(\"data/swissdock.wsdl\", \"w\") as f:\n",
+    "        f.write(WSDL)\n",
+    "    HERE = _dh[0]\n",
+    "    return Client(f\"file://{HERE}/data/swissdock.wsdl\")\n",
+    "\n",
+    "\n",
+    "def prepare_protein(client, protein):\n",
+    "    \"\"\"\n",
+    "    Given a PDB file (string contents), returns PSF and CRD\n",
+    "    \"\"\"\n",
+    "    encoded_protein = zlib.compress(protein.encode('utf-8'))\n",
+    "    job_id = client.service.prepareTarget(target=encoded_protein)\n",
+    "    while True:\n",
+    "        result = client.service.isTargetPrepared(jobID=job_id)\n",
+    "        if result is None:\n",
+    "            raise ValueError(\"No such a job present\")\n",
+    "        if result in (False, \"false\", 0):\n",
+    "            time.sleep(5)\n",
+    "        else:  # ready!\n",
+    "            break\n",
+    "    protein_files = client.service.getPreparedTarget(job_id)\n",
+    "    if protein_files is None or len(protein_files) != 2:\n",
+    "        raise ValueError(\"Could not prepare protein!\")\n",
+    "    return protein_files\n",
+    "            \n",
+    "\n",
+    "def prepare_ligand(client, ligand):\n",
+    "    \"\"\"\n",
+    "    Given a MOL2 file (string contents), returns PDB, RTF, PAR.\n",
+    "    \n",
+    "    Ligand must be protonated beforehand!\n",
+    "    \"\"\"\n",
+    "    encoded_ligand = zlib.compress(ligand.encode('utf-8'))\n",
+    "    job_id = client.service.prepareLigand(ligand=encoded_ligand)\n",
+    "    while True:\n",
+    "        result = client.service.isLigandPrepared(jobID=job_id)\n",
+    "        if result is None:\n",
+    "            raise ValueError(\"No such a job present\")\n",
+    "        if result in (False, \"false\", 0):\n",
```

Comment:
```suggestion
    "        if result in (False, \"false\", 1):\n",
```

Reply:
Closing this one because it was only an example :)